File: plugins/de.ovgu.featureide.examples/featureide_examples/Email-FH-AspectJ/features/Encrypt/EmailSystem/Email.java
package EmailSystem;

public class Email {
	protected boolean isEncrypted;
	protected int encryptionKey;
	
	boolean isReadable() {
		if (!isEncrypted())
			return original();
		else
			return false;
	}
	
	static void printMail(Email msg) {
		original(msg);
		Util.prompt("ENCRYPTED " + msg.isEncrypted());
		// Util.prompt("ENCRYPTION KEY  "+ msg.getEmailEncryptionKey());
	}
	
	boolean isEncrypted() {
		return isEncrypted;
	}


	void setEmailIsEncrypted(boolean value) {
		isEncrypted = value;
	}

	void setEmailEncryptionKey(int value) {
		this.encryptionKey = value;
	}

	int getEmailEncryptionKey() {
		return encryptionKey;
	}
}

File: plugins/de.ovgu.featureide.examples/featureide_examples/EmailSystem-FH-C/features/Encrypt/Email.c
void
printMail (struct email *msg)
{
  original (msg);
  printf ("ENCRYPTED\n  %i\n", msg->isEncrypted);
  printf ("ENCRYPTION KEY\n  %s\n", msg->encryptionKey);
}

struct email *
cloneEmail (struct email *msg)
{
  struct email *clone = original (msg);
  clone->isEncrypted = msg->isEncrypted;
  if (msg->encryptionKey)
    clone->encryptionKey = strdup (msg->encryptionKey);
  return clone;
}

int
isEncrypted (struct email *msg)
{
  return msg->isEncrypted;
}

int
isReadable (struct email *msg)
{
  if (0 == isEncrypted (msg))
    return original (msg);
  else
    return 0;
}
